src/main/java/ca/on/oicr/pineryreports/reports/impl/GazpachoProjectStatusReport.java

```diff
@@ -0,0 +1,617 @@
+
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+
+import ca.on.oicr.pinery.client.HttpResponseException;
+import ca.on.oicr.pinery.client.PineryClient;
+import ca.on.oicr.pineryreports.data.ColumnDefinition;
+import ca.on.oicr.pineryreports.reports.TableReport;
+import ca.on.oicr.pineryreports.util.CommonOptions;
+import ca.on.oicr.ws.dto.RunDto;
+import ca.on.oicr.ws.dto.RunDtoPosition;
+import ca.on.oicr.ws.dto.RunDtoSample;
+import ca.on.oicr.ws.dto.SampleDto;
+
+public class GazpachoProjectStatusReport extends TableReport {
+
+  private static class ReportItem implements Comparable<ReportItem> {
+
+    private final SampleDto stock;
+    private final String externalName;
+    private final Set<String> groupIds = new HashSet<>();
+    private final Set<SampleDto> aliquots = new HashSet<>();
+    private final Set<SampleDto> libraries = new HashSet<>();
+    private final Set<RunDto> runs = new HashSet<>();
+    private final Map<Integer, List<Integer>> lanesForRun = new HashMap<>();
+
+    public ReportItem(SampleDto stock, String externalName) {
+      this.stock = stock;
+      this.externalName = externalName;
+    }
+
+    public void addAliquot(SampleDto aliquot) {
+      this.aliquots.add(aliquot);
+    }
+
+    public void addLibrary(SampleDto library, String groupId) {
+      this.libraries.add(library);
+      if (groupId != null) groupIds.add(groupId);
+    }
+
+    public void addRunLane(RunDto run, RunDtoPosition lane) {
+      this.runs.add(run);
+      List<Integer> lanesForCurrentRun = lanesForRun.get(run.getId());
+      if (lanesForCurrentRun == null) {
+        List<Integer> lanes = new ArrayList<>();
+        lanes.add(lane.getPosition());
+        lanesForRun.put(run.getId(), lanes);
+      } else {
+        lanesForRun.get(run.getId()).add(lane.getPosition());
+      }
+    }
+
+    public SampleDto getStock() {
+      return stock;
+    }
+
+    public Set<SampleDto> getAliquots() {
+      return aliquots;
+    }
+
+    public String getWaitingAliquots() {
+      Date receivedByLab;
+      try {
+        receivedByLab = getDateTimeFormat().parse(stock.getCreatedDate());
+      } catch (java.text.ParseException e) {
+        System.out.println("Error parsing created date for stock " + stock.getId());
+        return "";
+      }
+      if (aliquots.isEmpty() && receivedByLab.toInstant().isBefore(TWO_DAYS_AGO)) {
+        return "1";
+      } else {
+        return "";
+      }
+    }
+
+    public String getWaitingLibraries() {
+      Date receivedByLab;
+      SampleDto mostRecentAliquot = aliquots.stream().max(Comparator.comparing(SampleDto::getCreatedDate)).orElse(null);
+      if (mostRecentAliquot == null) return "";
+      try {
+        receivedByLab = getDateTimeFormat().parse(mostRecentAliquot.getCreatedDate());
+      } catch (java.text.ParseException e) {
+        System.out.println("Error parsing created date for aliquot " + mostRecentAliquot.getId());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)